source/opt/type_manager.h

```diff
@@ -209,6 +209,8 @@ class TypeManager {
 
   IdToTypeMap id_to_incomplete_type_;  // Maps ids to their type representations
                                        // for incomplete types.
+
+  std::unordered_map<uint32_t, const Instruction*> id_to_constant_insn_;
```

Comment:
This looks like it is duplicating information that is in the constant manager.  Isn't there another way to use that instead?

Reply:
I am not entirely sure: the constant manager depends on the type manager (i.e. creating a constant requires passing in the corresponding type object), so the constant manager is built after the type manager. But now, for creating some of the types, we would need constants to already have been created, so we have a circular dependency.